packages/api-server/rest_server/tasks/dispatcher.py

```diff
@@ -0,0 +1,214 @@
+            return None, f"Missing Key value in json body: {ex}"
+        except Exception as ex:
+            return None, str(ex)
+        return req_msg, ""
+
+    def get_task_status(self):
+        """
+        Get all task status - This fn will trigger a ros srv call to acquire
+        all submitted tasks to dispatcher node. Fn returns an object of tasks
+        """
+        req = GetTaskList.Request()
+        try:
+            future = self.get_tasks_srv.call_async(req)
+            rclpy.spin_until_future_complete(self, future, timeout_sec=0.5)
+            response = future.result()
+            if response is None:
+                raise Exception("Unable to retrieve tasks")
+            else:
+                active_tasks = self.__convert_task_status_msg(
+                    response.active_tasks, False
+                )
+                terminated_tasks = self.__convert_task_status_msg(
+                    response.terminated_tasks, True
+                )
+                return {
+                    "active_tasks": active_tasks,
+                    "terminated_tasks": terminated_tasks,
+                }
+        except Exception as e:
+            self.get_logger().error("Error! GetTasks Srv failed %r" % (e,))
+        return []
+
+    def cancel_task_request(self, task_id) -> bool:
+        """
+        Cancel Task - This function will trigger a ros srv call to the
+        dispatcher node, and return a response.
+        """
+        req = CancelTask.Request()
+        req.task_id = task_id
+        try:
+            future = self.cancel_task_srv.call_async(req)
+            rclpy.spin_until_future_complete(self, future, timeout_sec=0.5)
+            response = future.result()
+            if response is None:
+                self.get_logger().warn("/cancel_task srv call failed")
+            else:
+                return response.success
+        except Exception as e:
+            self.get_logger().error("Error! Cancel Srv failed %r" % (e,))
+        return False
+
+    def __convert_task_status_msg(self, task_summaries, is_done=True):
+        """
+        convert task summary msg and return a jsonify-able task status obj
+        """
+        states_enum = {
+            0: "Queued",
+            1: "Active/Executing",
+            2: "Completed",
+            3: "Failed",
+            4: "Cancelled",
+            5: "Pending",
+        }
+        type_enum = {
+            0: "Station",
+            1: "Loop",
+            2: "Delivery",
+            3: "Charging",
+            4: "Clean",
+            5: "Patrol",
+        }
+
+        status_list = []
+        rclpy.spin_once(self, timeout_sec=0.0)
+        now = self.get_clock().now().to_msg().sec  # only use sec
+        for task in task_summaries:
+            desc = task.task_profile.description
+            status = {}
+            status["task_id"] = task.task_id
+            status["state"] = states_enum[task.state]
+            status["done"] = is_done
```

Comment:
doesn't the task state already contains information if it's done?

Reply:
I guess we can remove it if there is no need for the dashboard to explicitly display if the task is `done` or `not done` base on the definition above.